### PR TITLE
fix: added query to calc fees

### PR DESCRIPTION
**Pull request by bee344**
Partially #1365

### CODE REVIEW

File: src/services/blocks/BlocksService.ts [hunk 1]
```diff
@@ -168,6 +169,11 @@ export class BlocksService extends AbstractService {
 		const previousBlockHash = await this.fetchPreviousBlockHash(number);
 
 		for (let idx = 0; idx < block.extrinsics.length; ++idx) {
+			if (noFees === true) {
```
Comment (TarikGul) [inline src/services/blocks/BlocksService.ts:172]:
```suggestion
			if (noFees) {
```

**Review by marshacb — APPROVED** [review]
LGTM

**Comment by GCrispino** [comment]
Sorry to comment in a closed pull request, but if this was done to reduce performance issues, any specific reason for not making the loop that processes extrinsics concurrent?

**Comment by TarikGul** [comment]
@GCrispino There is a few things to tackle here:

1. This PR is aimed at providing an option to opt out of the fees. It's one of many ways we are going to increase the performance here.

2. https://github.com/paritytech/substrate-api-sidecar/pull/1367 We have a follow up PR here to increase the performance as well that does not require using a query parameter.

3. You are correct as we should find fees for each extrinsic concurrently moving forward, and we are working on getting a good solution going. But we also need to optimize the way we structure the calls to get the information necessary for fee calculation. We already have a PromiseQueue implemented for this: https://github.com/paritytech/substrate-api-sidecar/blob/master/src/util/PromiseQueue.ts#L5

Overall, we are taking steps to do this correctly. It's a balance between getting some fixes out now, as well as sorting out the larger issue correctly.

**Comment by GCrispino** [comment]
@TarikGul 

Got it! This makes sense, thanks for taking the time to explain 👍🏼

**Pull request merged**